lib/rules/filename-naming-convention.js

```diff
@@ -45,8 +46,24 @@ module.exports = {
   create(context) {
     return {
       Program: (node) => {
-        const rules = context.options[0];
-        const { ignoreMiddleExtensions } = context.options[1] || {};
+        const filenameWithPath = getFilePath(context);
+        const filename = getFilename(filenameWithPath);
+
+        const rules = transformRulesWithGroupCapture(
+          context.options[0],
+          filenameWithPath
+        );
+
```

Comment:
You get a error an error `There is an invalid pattern "featureA", please check it since it is not a glob or a builtin pattern` since you transform custom syntax firstly and then run into the `checkSettings` validation function. In this case, `<1>` will be `featureA`, but `featureA` is not a valid pattern.

I have an idea, we only allow three kinds of pattern styles for this rule: builtin pattern, glob pattern and our group capture syntax(`<index>` or `$index`). We only allow user choose one style at one time, and dont allow mix with them. So in this way, `<1>*`, `CAMEL_CASE*` both are illegal.

I feel like this would reduce the complexity, and make it more clear and easy to use.

Reply:
Yes it seems like a good idea.

I think I got most of it down but  I can't figure out what part of the coverage I'm missing.
I use sonar in some other projects to see which lines specifically aren't covered.
Do you use something similar ?

Reply:
> Yes it seems like a good idea.
> 
> I think I got most of it down but I can't figure out what part of the coverage I'm missing. I use sonar in some other projects to see which lines specifically aren't covered. Do you use something similar ?

Check out [this link](https://app.codecov.io/gh/DukeLuo/eslint-plugin-check-file/tree/group_capture).